libfs/fs.go

```diff
@@ -484,33 +486,81 @@ func (fs *FS) Readlink(link string) (target string, err error) {
 }
 
 // Chmod implements the billy.Filesystem interface for FS.
-func (fs *FS) Chmod(name string, mode os.FileMode) error {
-	return nil
+func (fs *FS) Chmod(name string, mode os.FileMode) (err error) {
+	fs.log.CDebugf(fs.ctx, "Chmod %s %s", name, mode)
+	defer func() {
+		fs.deferLog.CDebugf(fs.ctx, "Chmod done: %+v", err)
+	}()
+
+	n, _, err := fs.lookupOrCreateEntry(name, os.O_RDONLY, 0)
```

Comment:
I personally thought O_RDONLY was less confusing, since it makes it clear that the file wouldn't be created if it doesn't exist yet.  Keeping it that way for now, but if it becomes confusing we can change it later.